### ISSUE TITLE
Database errors in MySQL strict mode

### ISSUE THREAD

**Issue by willydee** [Froxlor/Froxlor#433]
### Problem

On a new Froxlor installation (Debian stretch), I've encountered error messages:
```
A database error occurred
SQLSTATE[HY000]: General error: 1364 Field 'destination' doesn't have a default value
```

### Steps to reproduce

- Set up a pristine host based on a recent Debian or Ubuntu distribution
- Verify that MySQL is running in strict mode
- Create a customer and domain
- As customer, try to create a new email address

### Possible cause

MySQL is operating in strict mode by default (in Debian jessie or newer), which causes errors if a `text` type column declared `NOT NULL` is omitted in an `INSERT` operation. See https://dev.mysql.com/doc/refman/5.7/en/data-type-defaults.html.

### Suggestions
- Declare `text` type columns without `NOT NULL` constraint, or
- Use sufficiently large `varchar` type columns instead of `text`, or
- Supply empty string values for the affected columns when inserting

RFC. I'll supply a pull request once it is decided which solution should be applied.

Best regards, willydee

**Comment by d00p** [comment]
Being strict-mode compatible would be the best choice as also ubuntu is using this mode by default now; a pull request - tested! - would be most helpful

**Comment by Starcalc** [comment]
During installation on a new ubuntu (17.04) with mysql 5.7, this fails even during installation. Yet worse, the installation sql-script does not check whether the imported sql-statements have failed. This results in various problems, e.g. the "mail_virtual"-table is never created due to an error. The installation website results with everything green although an error has occured.

When I try to create the mail_virtual-table in CLI-mode of mysql, I get a more explanatory error which pointed me to this issue.

```
mysql> CREATE TABLE `mail_virtual` (
    ->   `id` int(11) NOT NULL auto_increment,
    ->   `email` varchar(255) NOT NULL default '',
    ->   `email_full` varchar(255) NOT NULL default '',
    ->   `destination` text NOT NULL default '',
    ->   `domainid` int(11) NOT NULL default '0',
    ->   `customerid` int(11) NOT NULL default '0',
    ->   `popaccountid` int(11) NOT NULL default '0',
    ->   `iscatchall` tinyint(1) unsigned NOT NULL default '0',
    ->   PRIMARY KEY  (`id`),
    ->   KEY `email` (`email`)
    -> ) ENGINE=MyISAM CHARSET=utf8 COLLATE=utf8_general_ci;
ERROR 1101 (42000): BLOB, TEXT, GEOMETRY or JSON column 'destination' can't have a default value

```

**Comment by d00p** [comment]
@Starcalc known, as always: froxlor is not yet compatible with strict-mode, also does not yet officially support 16.06/17.04...

**Comment by d00p** [comment]
Might be fixed due to 1e03946df7e0a8aebe8d2c680b9349be81dafda3
